src/transforms/sort.js

```diff
@@ -0,0 +1,189 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Lib = require('../lib');
+var PlotSchema = require('../plot_api/plot_schema');
+var axisIds = require('../plots/cartesian/axis_ids');
+var autoType = require('../plots/cartesian/axis_autotype');
+var setConvert = require('../plots/cartesian/set_convert');
+
+exports.moduleType = 'transform';
+
+exports.name = 'sort';
+
+exports.attributes = {
+    enabled: {
+        valType: 'boolean',
+        dflt: true,
+        description: [
+            'Determines whether this sort transform is enabled or disabled.'
+        ].join(' ')
+    },
+    target: {
+        valType: 'string',
+        strict: true,
+        noBlank: true,
+        arrayOk: true,
+        dflt: 'x',
+        description: [
+            'Sets the target by which the sort transform is applied.',
+
+            'If a string, *target* is assumed to be a reference to a data array',
+            'in the parent trace object.',
+            'To sort about nested variables, use *.* to access them.',
+            'For example, set `target` to *marker.size* to sort',
+            'about the marker size array.',
+
+            'If an array, *target* is then the data array by which',
+            'the sort transform is applied.'
+        ].join(' ')
+    },
+    order: {
+        valType: 'enumerated',
+        values: ['ascending', 'descending'],
```

Comment:
I chose _ascending_ and `descending` to be familiar with axis [`categoryorder`](https://github.com/plotly/plotly.js/blob/ef0946cfef97c446363edc11edf2e149d9297438/src/plots/cartesian/layout_attributes.js#L578)